### PR TITLE
PSMDB-916 Doc restructure

### PR DESCRIPTION
**Pull request by nastena1606**
- Separated vault integration and keyfile management into separate docs
- Created a How to section
- Separated SASL auth configuration and moved it  to How to
- Added diagrams for external auth methods

### CODE REVIEW

File: source/sasl-auth.rst [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+Configuring ``saslauthd``
+---------------------------------
+1. Install the SASL packages. Depending on your OS, use the following command:
+
+   For *RedHat and CentOS*:
+
+   .. code-block:: bash
+
+      $ sudo yum install -y cyrus-sasl cyrus-sasl-plain
```
Comment (sandraromanchenko) [inline source/sasl-auth.rst]: For 4.0.26, 4.2.16, 4.4.8, 5.0.2 (PSMDB-873) cyrus-sasl-plain package isn't required, it's installed as a dependency

File: source/sasl-auth.rst [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+   ldap_servers: ldap://localhost
+   ldap_mech: PLAIN
+   ldap_search_base: dc=example,dc=com
+   ldap_filter: (cn=%u)
+   ldap_bind_dn: cn=admin,dc=example,dc=com
+   ldap_password: secret
+
+Note the LDAP password (``ldap_password``) and bind domain name (``ldap_bind_dn``).
+This allows the ``saslauthd`` service to connect to the LDAP service as root.
```
Comment (sandraromanchenko) [inline source/sasl-auth.rst]: it's not clear why these settings allow to connect as root, I'd remove this statement at all.

File: source/sasl-auth.rst [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+The ``log_level`` is optional but may help determine configuration errors.
+
+.. seealso::
+
+   `SASL documentation: <https://www.cyrusimap.org/sasl/index.html>`_
+  
+Configuring ``mongod`` Server
+-----------------------------
+
+To enable external authentication, you must create a user with the **root** privileges in the ``admin`` database. If you have already created this user, skip this step. Otherwise, run the following command to create the admin user:
```
Comment (sandraromanchenko) [inline source/sasl-auth.rst]: These is no need to create root user for external auth, however enabling access control (authorization: enabled) enforces authentication, so user is simply required for further actions with DB.

File: source/install/uninstall.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
+.. _uninstall:
+
+Uninstalling |PSMDB|
+********************
+
+To completely remove |PSMDB| you need to remove all the installed packages, data and configuration files. f you need the data, consider making a backup before uninstalling Percona Server for MongoDB.
```
Comment (sandraromanchenko) [inline source/install/uninstall.rst]: Misprint - IF you need data..

File: source/authentication.rst [hunk 5]
```diff
@@ -28,9 +28,13 @@ LDAP authentication using SASL
    :local:
    :depth: 1
 
+
 Overview
 ========
 
+LDAP authentication using |SASL| means that both the client and the server establish a SASL session using the SASL library. Then authentication (bind) requests are sent to the LDAP server through the SASL authentication daemon (``saslauthd``) that acts as a remote proxy for the ``mongod`` server.
```
Comment (sandraromanchenko) [inline source/authentication.rst]: I'd rephrase this - it's not LDAP auth using SASL, it's rather auth VIA saslauthd daemon or LDAP proxy authentication (as mongodb official doc suggests)

File: source/authentication.rst [hunk 6]
```diff
@@ -67,242 +71,18 @@ An authentication session uses the following sequence:
...
-  
+For configuration instructions, refer to :ref:`sasl`.
 
 .. _ldap-authorization:
 
 Authentication and Authorization with Direct Binding to LDAP
 ============================================================
-  
+
 This feature has been supported in MongoDB Enterprise since its version 3.4.
```
Comment (sandraromanchenko) [inline source/authentication.rst]: Is it possible also to add version when this feature appeared in PSMDB?

**Review by sandraromanchenko — CHANGES_REQUESTED** [review]

File: source/authentication.rst (hunk 6 above)
Comment (nastena1606) [reply source/authentication.rst]: Added

File: source/install/uninstall.rst (hunk 4 above)
Comment (nastena1606) [reply source/install/uninstall.rst]: Fixed

File: source/sasl-auth.rst (hunk 3 above)
Comment (nastena1606) [reply source/sasl-auth.rst]: So what role is sufficient for such a user?

File: source/authentication.rst (hunk 5 above)
Comment (nastena1606) [reply source/authentication.rst]: Done

File: source/sasl-auth.rst (hunk 2 above)
Comment (nastena1606) [reply source/sasl-auth.rst]: @igorsol Please kindly assist here, shall we remove this statement or how can we rephrase it to explain what's going on here?

Comment (igorsol) [reply source/sasl-auth.rst]: Probably the root is mentioned because example `ldap_bind_dn` is admin? The next line clearly says "users should not store administartive passwords...".
I would rephrase this by replacing "as root" with "as administrator".

**Review by sandraromanchenko — APPROVED** [review]

File: source/sasl-auth.rst [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+
+   .. code-block:: yaml
+
+      MECH=ldap		
+
+   For *Debian/Ubuntu*, use the following commands to enable the ``saslauthd`` to auto-run on startup and to set the ``ldap`` value for the ``--MECHANISMS`` option:
+
+   .. code-block:: bash
+
+      $ sudo sed -i -e s/^MECH=pam/MECH=ldap/g /etc/sysconfig/saslauthdsudo sed -i -e s/^MECHANISMS="pam"/MECHANISMS="ldap"/g /etc/default/saslauthdÂ
```
Comment (igorsol) [inline source/sasl-auth.rst]: Is this Â intended or some artifact?

File: source/sasl-auth.rst [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+   ldap_mech: PLAIN
+   ldap_search_base: CN=Users,DC=example,DC=com
+   ldap_filter: (sAMAccountName=%ucn=%u)
+   ldap_bind_dn: CN=ldapmgr,CN=Users,DC=<AD Domain>,DC=<AD TLD>
+   ldap_password: ld@pmgr_Pa55word
+
+In order to determine and test the correct search base
+and filter for your Active Directory installation,
+the Microsoft `LDP GUI Tool
+<https://technet.microsoft.com/en-us/library/Cc772839%28v=WS.10%29.aspx>`_
```
Comment (igorsol) [inline source/sasl-auth.rst:130]: Is underscore at the end of line intended?

**Review by igorsol — APPROVED** [review]
